crates/forge/src/scarb.rs

```diff
@@ -226,6 +237,38 @@ pub fn dependencies_for_package(
     Ok(dependencies)
 }
 
+fn validate_fork_config(config: Result<ForgeConfig>) -> Result<ForgeConfig> {
+    if let Ok(ForgeConfig { fork: forks, .. }) = &config {
+        let names: Vec<String> = forks.iter().map(|fork| fork.name.clone()).collect();
+        let removed_duplicated_names: Vec<String> = names.clone().into_iter().unique().collect();
+
+        if names.len() != removed_duplicated_names.len() {
+            return Err(anyhow!("Some fork names are duplicated"));
+        }
+
+        for fork in forks {
+            let block_id_items: Vec<(&String, &String)> = fork.block_id.iter().collect();
+            let [(block_id_key, block_id_value)] = block_id_items[..] else {
+                return Err(anyhow!("block_id should be set once per fork"));
+            };
+
+            if !["number", "hash", "tag"].contains(&&**block_id_key) {
+                return Err(anyhow!(
+                    "block_id has only three variants: number, hash and tag"
+                ));
+            }
+
+            if block_id_key == "tag" && !["Latest", "Pending"].contains(&&**block_id_value) {
```

Comment:
`&&**` <- what's that?

Reply:
@war-in?

Reply:
`["Latest", "Pending"]` is of type `[&str;2]`. [`contains`](https://doc.rust-lang.org/std/primitive.slice.html#method.contains) expects a reference to its element, i. e. `&&str`. `block_id_value` is of type `&&String`, therefore to get `str` we first do double deref and then to get `&&str` double ref. I guess Rust cannot handle it itself, this post provides deeper explanation but didn't dig much into it: https://stackoverflow.com/questions/28519997/what-are-rusts-exact-auto-dereferencing-rules